data/org.freedesktop.portal.GameMode.xml

```diff
@@ -0,0 +1,78 @@
+<?xml version="1.0"?>
+<!--
+ Copyright (C) 2019 Red Hat, Inc.
+ This library is free software; you can redistribute it and/or
+ modify it under the terms of the GNU Lesser General Public
+ License as published by the Free Software Foundation; either
+ version 2 of the License, or (at your option) any later version.
+ This library is distributed in the hope that it will be useful,
+ but WITHOUT ANY WARRANTY; without even the implied warranty of
+ MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ Lesser General Public License for more details.
+ You should have received a copy of the GNU Lesser General Public
+ License along with this library. If not, see <http://www.gnu.org/licenses/>.
+ Author: Christian J. Kellner <christian@kellner.me>
+-->
+
+<node name="/" xmlns:doc="http://www.freedesktop.org/dbus/1.0/doc.dtd">
+    <!--
+      org.freedesktop.portal.GameMode:
+      @short_description: Portal for accessing GameMode
+
+      Interface for accessing GameMode from within the sandbox. It
+      is analogous to the com.feralinteractive.GameMode interface
+      and will proxy request there, but with permission checking and
+      pid mapping (from the sandbox' pid namespace to the host one)
+      if necessary.
+
+      This documentation describes version 1 of this interface.
+    -->
+    <interface name="org.freedesktop.portal.GameMode">
+      <!--
+          QueryStatus:
+          @pid: Process id to query the GameMode status of
+          @result: The GameMode status for @pid
+
+          Query the GameMode status for a process. Will return:
+            0 if GameMode is inactive,
+            1 if GameMode is active,
+            2 if GameMode is active and @pid is registered,
+           -1 if the query failed inside GameMode
+      -->
+      <method name="QueryStatus">
+        <arg type="i" name="pid" direction="in"/>
+        <arg type="i" name="result" direction="out"/>
+      </method>
+      <!--
+          RegisterGame:
+          @pid: Process id of the game to register
+          @result: Status of the request
+
+          Register a game with GameMode and thus request GameMode to
+          be activated. Will return:
+            0 if the game with @pid was successfully registered,
+           -1 if the request was rejected by GameMode
+      -->
+      <method name="RegisterGame">
+        <arg type="i" name="pid" direction="in"/>
+        <arg type="i" name="result" direction="out"/>
+      </method>
+      <!--
+          UnregisterGame:
+          @pid: Process id of the game to un-register
+          @result: Status of the request
+
+          Un-register a game from GameMode; if the call is successful
+          and @pid was the last registered game, GameMode will be
```

Comment:
The last registered game? This is unclear - isn't this just about how often Register/Unregister have been called for the same pid ?

Related question: what happens when the caller exits without unregistering ?

Reply:
What I mean was: if the game with `pid` was the only game left that requested GameMode then GameMode will be globally deactivated; conversely even if `Unregister` ist successfully done for `pid` there might be other clients/games registered so GameMode will stay active.

Reply:
> Related question: what happens when the caller exits without unregistering ?

The GameMode session daemon periodically [checks](https://github.com/FeralInteractive/gamemode/blob/e6c90f4de1e5e017ff2712c34e1c59ab80de5f07/daemon/gamemode.c#L283) if `pid` still exists, and auto-unregisters "expired" clients/games.

Reply:
I rephrased it, maybe it is clearer now? I also added some bits about monitoring to the general introduction.